controllers/challenge.js

```diff
@@ -12,10 +12,17 @@ const sendChallengeResponse = async (req, res) => {
   try {
     if (req.method === 'GET') {
       const allChallenges = await challengeQuery.fetchChallenges()
-      if (allChallenges.length > 0) {
+      const newChallenges = await Promise.all(allChallenges.map(async (challenge) => {
+        const getParticipants = await challengeQuery.fetchParticipants(challenge.participants)
```

Comment:
If there are 1,000 participants
You're to going to make this loop 1,000 times
Internally each loop is going to 1000/10 = 100 times

You'll be running it 1,000,00 times?? 

Isn't that highly inefficient?